app/controllers/repositories_controller.rb

```diff
@@ -52,7 +52,9 @@ def list
   end
 
   def rows_list
-    render json: { data: @repository.repository_rows.map { |r| [r.id, r.name] } }
+    results = @repository.repository_rows
+    results = results.name_like(params[:query]) if params[:query].present?
+    render json: { data: results.map { |r| [r.id, r.name] } }
```

Comment:
I think here you can just do `results = @repository.repository_rows.select(:id, :name)` in the begining and then you don't need the map.